pkg/standalone-utils/test/SiloWrapping.test.ts

```diff
@@ -0,0 +1,811 @@
+              value: expectedsDAIAmount,
+            },
+            sDAI
+          );
+        });
+      }
+    });
+
+    describe('unwrap sDAI', () => {
+      let tokenSender: Account, tokenRecipient: Account;
+
+      context('sender = senderUser, recipient = relayer', () => {
+        beforeEach(async () => {
+          await sDAI.connect(senderUser).approve(vault.address, fp(10));
+          tokenSender = senderUser;
+          tokenRecipient = relayer;
+        });
+        testUnwrap();
+      });
+
+      context('sender = senderUser, recipient = senderUser', () => {
+        beforeEach(async () => {
+          await sDAI.connect(senderUser).approve(vault.address, fp(10));
+          tokenSender = senderUser;
+          tokenRecipient = senderUser;
+        });
+        testUnwrap();
+      });
+
+      context('sender = relayer, recipient = relayer', () => {
+        beforeEach(async () => {
+          await sDAI.connect(senderUser).transfer(relayer.address, amount);
+          tokenSender = relayer;
+          tokenRecipient = relayer;
+        });
+        testUnwrap();
+      });
+
+      context('sender = relayer, recipient = senderUser', () => {
+        beforeEach(async () => {
+          await sDAI.connect(senderUser).transfer(relayer.address, amount);
+          tokenSender = relayer;
+          tokenRecipient = senderUser;
+        });
+        testUnwrap();
+      });
+
+      function testUnwrap(): void {
+        beforeEach('Load the liquidity pool with underlying tokens', async () => {
+          await DAI.connect(senderUser).mint(mockSilo.address, amount);
+        });
+
+        it('unwraps with immediate amounts', async () => {
+          const receipt = await (
+            await relayer.connect(senderUser).multicall([encodeUnwrap(tokenSender, tokenRecipient, amount)])
+          ).wait();
+          const relayerIsSender = TypesConverter.toAddress(tokenSender) === relayer.address;
+
+          if (!relayerIsSender) {
+            expectTransferEvent(
+              receipt,
+              {
+                from: TypesConverter.toAddress(senderUser),
+                to: TypesConverter.toAddress(relayer),
+                value: amount,
+              },
+              sDAI
+            );
+          }
+          expectTransferEvent(
+            receipt,
+            {
+              from: TypesConverter.toAddress(relayer),
+              to: ZERO_ADDRESS,
+              value: amount,
+            },
+            sDAI
+          );
+
+          const relayerIsRecipient = TypesConverter.toAddress(tokenRecipient) === relayer.address;
+          expectTransferEvent(
+            receipt,
+            {
+              from: TypesConverter.toAddress(relayerIsRecipient ? mockSilo : relayer),
+              to: TypesConverter.toAddress(relayerIsRecipient ? relayer : senderUser),
+              value: amount,
+            },
+            DAI
+          );
```

Comment:
This should actually be two steps:
1. **always**: `mockSilo -> relayer`
2. **if relayer is not recipient**: `relayer -> tokenSender` **(NOT senderUser)**